### DISCUSSION TITLE
Statistics dashboard for Bundler, Ruby, and Rubygems versions

### DISCUSSION THREAD

**Discussion by coilysiren** [rubygems/rubygems.org#3350] [Ideas]
(migrated from the Trello board)

> Thanks to André Mediros, we are collecting Bundler, Ruby, and Rubygems versions in Librato. It would be great to have a dashboard that shows just the versions of Bundler, Rubygems, and Ruby that are being used, so we can see how many users we will be affecting with future compatibility decisions.
> 
> This is an example of platform and version statistics that are super useful, maybe something like this? http://sparkle.psionides.eu/feeds/gitifier/statistics
> 
> We have some partial graphs on the main bundler-api Librato dashboard, but they need to be updated for new versions of Ruby and Bundler. It would also be great to see graphs of how often install is run vs update, as well as graphs of OS, platform, RubyGems version, config settings, and the other data we are collecting.
> 
> We also collect CI system names now, it would be cool to have a graph of CI vs non-CI commands as well as a breakdown of CI system percentages.

**Comment by dwradcliffe** [comment]
The existing stats are totally broken now that we're using Fastly and we're also moving away from Librato. This would be neat but at the moment I don't see how it's going to happen.

**Comment by segiddins** [comment]
@dwradcliffe it'd need to process fastly logs, just like the stats

**Comment by dwradcliffe** [comment]
Yep, but that's a LOT more requests to process

**Comment by jaredbeck** [comment]
Would gem authors be able to see statistics about, e.g. what version of ruby their users have? This would be very useful to gem authors when deciding which versions of ruby to support. Many gems also wonder which versions of rails they should support. Sinatra plugins will want to know what versions of sinatra are in use, etc. Giving gem authors the data to make informed decisions about compatibility would be beneficial to the whole ruby community.

**Comment by simi** [comment]
There are some global stats at http://ecosystem.rubytogether.org/. Currently there are no stats for individual Gems available. Moving this into discussions for now.
